tests/src/Common/runincontext/runincontext.cs

```diff
@@ -0,0 +1,562 @@
+                    Console.WriteLine($"    + Average leak over the last {i} iterations: {leak} bytes\n");
+
+                    restarts++;
+                    if (restarts == _input.MaxRestarts)
+                    {
+                        break;
+                    }
+                    i = 0;
+                    startMemory = currentMemory;
+                    leak = 0;
+                    lastProgressReport = 0;
+                    continue;
+                }
+            }
+
+            if ((i == 2) && (startSpeed == 0))
+            {
+                startSpeed = currentSpeed;
+            }
+        }
+
+        //sometimes this happens (no real reason, but it's not a failure, so let's not write a "negative" leak to the output)
+        if (currentMemory < startMemory)
+        {
+            startMemory = currentMemory;
+            leak = 0;
+        }
+        if (_input.MonitorMode)
+        {
+            leak = (long)((currentMemory - monitorStartMem) / ((_input.StressModeCount * 1.0) - _input.IterationsToSkip));
+            startMemory = monitorStartMem;
+        }
+
+        Console.WriteLine("\n==================================================");
+        Console.WriteLine($"Starting memory size          : {startMemory / 1024} KB");
+        Console.WriteLine($"Ending memory size            : {currentMemory / 1024} KB");
+        Console.WriteLine();
+        Console.WriteLine($"Starting emission speed       : {startSpeed} milliseconds");
+        Console.WriteLine($"Ending emission speed         : {currentSpeed} milliseconds");
+        Console.WriteLine();
+        Console.WriteLine($"Memory leak                   : {currentMemory - startMemory} bytes ({leak} bytes per iteration).");
+        if (currentMemory > startMemory)
+        {
+            throw new Exception("Memory leaked");
+        }
+
+        return retVal;
+    }
+
+    public int ExecuteAssemblyEntryPoint(MethodInfo entryPoint)
+    {
+        int result = 0;
+
+        object res;
+        object[] args = (entryPoint.GetParameters().Length != 0) ? new object[] { _input.EntryArgs } : null;
+        string argsStr = (args == null) ? "" : _input.EntryArgStr.ToString();
+
+        if (_input.Verbose)
+        {
+            Console.WriteLine($"Invoking Main({argsStr})\n");
+        }
+
+        res = entryPoint.Invoke(null, args);
+
+        result = (entryPoint.ReturnType == typeof(void)) ? Environment.ExitCode : Convert.ToInt32(res);
+
+        return result;
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public int ExecuteAndUnload(List<WeakReference> assemblyReferences, out WeakReference testAlcWeakRef, out WeakReference testAlcWeakRefInner)
+    {
+        int result;
+        TestAssemblyLoadContextBase testAlc = new TestAssemblyLoadContext(_input.AssemblyPath, _input.ReferencesPath, assemblyReferences);
+
+        if (_input.DelegateLoad)
```

Comment:
What is DelegateLoad option for - when would one use it?

Reply:
It is a testing mode where there are two unloadable AssemblyLoadContexts where the main context contains only the test executable and the other context contains all of its dependencies. There were one or two bugs in the runtime that were uncovered by this mode that I have fixed in the past.